lib/vquic/quiche.c

```diff
@@ -681,6 +681,9 @@ static ssize_t h3_stream_send(struct Curl_easy *data,
     H3BUGF(infof(data, "Pass on %zd body bytes to quiche", len));
     sent = quiche_h3_send_body(qs->h3c, qs->conn, stream->stream3_id,
                                (uint8_t *)mem, len, FALSE);
+    if(sent == QUICHE_H3_ERR_DONE) {
+      sent = len;
```

Comment:
This should be `sent = 0` or something like that, since the `QUICHE_H3_ERR_DONE` error code means that no data was sent, and the write should be retried later.

Reply:
yeah, seem you're right here, thx !

Reply:
seem that i got strange result when setting it to 0

Reply:
so, in fact, i really wonder if quiche must not return BufferTooShort instead of Done here:
https://github.com/cloudflare/quiche/blob/master/quiche/src/h3/mod.rs#L1037